### PR TITLE
Paas 1786

### PR DESCRIPTION
**Pull request by Ronan-Gorain**
JIRA issue: https://jira.jahia.org/browse/PAAS-1786

Short description:

### CODE REVIEW

File: mixins/jahia.yml [hunk 1]
```diff
@@ -854,6 +858,133 @@ actions:
...
+
+  setPatInJahia:
+    # Parameters:
+    # __secret__pat_token: the personal access token
+    # jahia_version: the jahia version
+    - cmd[proc]: |-
+        __secret__pat_token=${this.__secret__pat_token}
+        groovy_file_path=/data/digital-factory-data/patches/groovy/pat.groovy
+
+        jahia_major_version=$(echo "${this.jahia_version}" | head -c1)
```
Comment (laurentfufu) [inline mixins/jahia.yml:913]: looks dangerous (although very unlikely), but I will prefer something like this:
```suggestion
        jahia_major_version=$(echo "${this.jahia_version}" | cut -d'.' -f1)
```

File: mixins/jahia.yml [hunk 2]
```diff
@@ -854,6 +858,133 @@ actions:
...
+        echo """
+        org.jahia.services.content.JCRTemplate.getInstance().doExecuteWithSystemSession({ session ->
+          org.jahia.osgi.BundleUtils.getOsgiService(\"org.jahia.modules.apitokens.TokenService\"$jahia_7_parameter)
+              .tokenBuilder(\"/users/root\", \"Jahia Cloud Token\", session)
+              .setToken(\"$__secret__pat_token\")
+              .setActive(true)
+              .create()
+          session.save();
+        })
+        """ >> $groovy_file_path
```
Comment (laurentfufu) [inline mixins/jahia.yml:933]: from the Avoid Backslash Comity:
```suggestion
        echo '''
        org.jahia.services.content.JCRTemplate.getInstance().doExecuteWithSystemSession({ session ->
          org.jahia.osgi.BundleUtils.getOsgiService("org.jahia.modules.apitokens.TokenService"$jahia_7_parameter)
              .tokenBuilder("/users/root", "Jahia Cloud Token", session)
              .setToken("$__secret__pat_token")
              .setActive(true)
              .create()
          session.save();
        })
        ''' >> $groovy_file_path
```

File: packages/jahia/migrations/migrate-to-v13.yaml [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+    - getJahiaVersion
+    - enableKarafLogin: proc
+    - removeHealthcheckModule
+
+    - isVersionHigherOrEqual:
+        a: ${globals.jahiaVersion}
+        b: 8.1.0.0
+        res: jahia8
+
+    - if ( ${globals.jahia8}):
```
Comment (laurentfufu) [inline packages/jahia/migrations/migrate-to-v13.yaml:78]:
```suggestion
    - if (${globals.jahia8}):
```

File: packages/jahia/migrations/migrate-to-v13.yaml [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+        - isVersionStrictlyLower:
+            a: ${globals.jahiaVersion}
+            b: 8.1.0.0
+            res: jahia80
+        - if ( ${globals.jahia80}):
```
Comment (laurentfufu) [inline packages/jahia/migrations/migrate-to-v13.yaml:83]:
```suggestion
        - if (${globals.jahia80}):
```

File: packages/jahia/migrations/migrate-to-v13.yaml [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+
+        - installModuleVersionAndRemoveOthers:
+            moduleName: ${this.patModuleName}
+            version: 0.5.0
+        - installModuleVersionAndRemoveOthers:
+            moduleName: ${this.samModuleName}
+            version: ${this.samVersion}
+
+    - cmd[cp, proc]: |-
+        echo "probes.PatchFailures.severity=HIGH" >> /data/digital-factory-data/karaf/etc/org.jahia.modules.sam.healthcheck.ProbesRegistry.cfg
```
Comment (laurentfufu) [inline packages/jahia/migrations/migrate-to-v13.yaml:117]: Can it lead to duplicated line if migration is relaunched ?
Isn't this conf file replicated between jahia cluster ?

File: packages/jahia/migrations/migrate-to-v13.yaml [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+
+    - getVaultData
+    - vaultDoesSecretExistsOrIsReadable:
+        secretPath: "paas/customers/${globals.organizationName}/paas_${env.shortdomain}/PAT/root"
+    - if ( ! ${globals.secretExistsOrIsReadable}):
```
Comment (laurentfufu) [inline packages/jahia/migrations/migrate-to-v13.yaml:122]: _please select:_
```suggestion
    - if (! ${globals.secretExistsOrIsReadable}):
```
_or_
```suggestion
    - if ( ! ${globals.secretExistsOrIsReadable} ):
```
:trollface:

File: packages/jahia/migrations/migrate-to-v13.yaml [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+          echo "Healthcheck module is still installed"
+          exit 1
+        fi
+
+  jahiaLegacyCheckHealth:
+    - cmd [${nodes.bl.first.id}]: |-
+        if [ ! -z "$(grep ping.jsp /etc/haproxy/haproxy.cfg)" ]; then
+          echo "Migration has already been ran unsuccessfully on this env, healthcheck is in an unknown state"
+        fi
+    - if ("${response.out}" == "" ):
```
Comment (laurentfufu) [inline packages/jahia/migrations/migrate-to-v13.yaml:266]:
```suggestion
    - if ("${response.out}" == ""):
```

Comment (Ronan-Gorain) [reply packages/jahia/migrations/migrate-to-v13.yaml]: done

File: packages/jahia/migrations/migrate-to-v13.yaml (hunk 6 above)
Comment (Ronan-Gorain) [reply packages/jahia/migrations/migrate-to-v13.yaml]: done

File: packages/jahia/migrations/migrate-to-v13.yaml (hunk 5 above)
Comment (Ronan-Gorain) [reply packages/jahia/migrations/migrate-to-v13.yaml]: done

File: packages/jahia/migrations/migrate-to-v13.yaml (hunk 4 above)
Comment (Ronan-Gorain) [reply packages/jahia/migrations/migrate-to-v13.yaml]: done

File: packages/jahia/migrations/migrate-to-v13.yaml (hunk 3 above)
Comment (Ronan-Gorain) [reply packages/jahia/migrations/migrate-to-v13.yaml]: done

File: mixins/jahia.yml (hunk 2 above)
Comment (Ronan-Gorain) [reply mixins/jahia.yml]: done

File: mixins/jahia.yml (hunk 1 above)
Comment (Ronan-Gorain) [reply mixins/jahia.yml]: done

**Review by laurentfufu — APPROVED** [review]

**Pull request merged**